### PR TITLE
Persist failpoints across member restart

### PR DESCRIPTION
**Pull request by jpbetz**
Fix https://github.com/etcd-io/etcd/issues/11898

Noticed that if etcd is restarted, we lose track of any gofail failpoints. This PR persists them across restarts.

### CODE REVIEW

**Comment by gyuho** [comment]
lgtm, thanks for the fix!

**Comment by jingyih** [comment]
LGTM. Thanks!

File: functional/agent/handler.go [hunk 1]
```diff
@@ -109,11 +109,15 @@ func (srv *Server) creatEtcd(fromSnapshot bool) error {
...
 		zap.String("etcd-exec", etcdPath),
 		zap.Strings("etcd-flags", etcdFlags),
+		zap.String("GOFAIL_FAILPOINTS", failpoints),
 		zap.String("failpoint-http-addr", srv.Member.FailpointHTTPAddr),
 		zap.String("failpoint-addr", u.Host),
 	)
 	srv.etcdCmd = exec.Command(etcdPath, etcdFlags...)
 	srv.etcdCmd.Env = []string{"GOFAIL_HTTP=" + u.Host}
+	if failpoints != "" {
+		srv.etcdCmd.Env = append(srv.etcdCmd.Env, "GOFAIL_FAILPOINTS=" + failpoints)
```
Comment (jingyih) [inline functional/agent/handler.go:119]: NIT: gofmt is complaining this line
```
-		srv.etcdCmd.Env = append(srv.etcdCmd.Env, "GOFAIL_FAILPOINTS=" + failpoints)
+		srv.etcdCmd.Env = append(srv.etcdCmd.Env, "GOFAIL_FAILPOINTS="+failpoints)
```

**Comment by gyuho** [comment]
Manually fixed fmt.

**Comment by jpbetz** [comment]
Thanks @gyuho! I forgot about this one somehow in everything else going on. Appreciate you getting it merged.

**Pull request merged**